creator-node/src/services/stateMachineManager/stateMonitoring/findSyncRequests.jobProcessor.js

```diff
@@ -290,60 +274,58 @@ const _findSyncsForUser = async (
       errors.push(
         `Error computing sync mode for user ${wallet} and secondary ${secondary} - ${e.message}`
       )
       continue
     }
 
-    if (syncMode === SYNC_MODES.SyncSecondaryFromPrimary) {
+    let result
+    if (syncMode === SYNC_MODES.None) {
+      result = 'no_sync_secondary_data_matches_primary'
+    } else if (
+      syncMode === SYNC_MODES.SyncSecondaryFromPrimary ||
+      syncMode === SYNC_MODES.MergePrimaryAndSecondary
+    ) {
       try {
         const { duplicateSyncReq, syncReqToEnqueue } = getNewOrExistingSyncReq({
           userWallet: wallet,
           primaryEndpoint: thisContentNodeEndpoint,
           secondaryEndpoint: secondary,
-          syncType: SyncType.Recurring
+          syncType: SyncType.Recurring,
+          syncMode
         })
 
         if (!_.isEmpty(syncReqToEnqueue)) {
-          resultBySecondary[secondary] =
-            'new_sync_request_enqueued_primary_to_secondary'
+          result = 'new_sync_request_enqueued_primary_to_secondary'
           syncReqsToEnqueue.push(syncReqToEnqueue)
         } else if (!_.isEmpty(duplicateSyncReq)) {
-          resultBySecondary[secondary] = 'sync_request_already_enqueued'
+          result = 'sync_request_already_enqueued'
           duplicateSyncReqs.push(duplicateSyncReq)
         } else {
-          resultBySecondary[secondary] = 'new_sync_request_unable_to_enqueue'
+          result = 'new_sync_request_unable_to_enqueue'
         }
       } catch (e) {
-        resultBySecondary[secondary] = 'no_sync_unexpected_error'
+        result = 'no_sync_unexpected_error'
         errors.push(
-          `Error getting new or existing sync request for user ${wallet} and secondary ${secondary} - ${e.message}`
+          `Error getting new or existing sync request for syncMode ${syncMode}, user ${wallet} and secondary ${secondary} - ${e.message}`
         )
-        continue
       }
-    } else if (syncMode === SYNC_MODES.MergePrimaryAndSecondary) {
-      /**
-       * TODO - currently just logs as placeholder
-       * 1. Primary will sync all content from secondary
-       * 2. Primary will force secondary to wipe its local state and resync all content
-       */
-      logger.info(
-        `[findSyncRequests][_findSyncsForUser][MergePrimaryAndSecondary = true][SyncType = ${SyncType.Recurring}] wallet ${wallet} secondary ${secondary} Clocks: [${primaryClock},${secondaryClock}] Files hashes: [${primaryFilesHash},${secondaryFilesHash}]`
-      )
+    }
 
-      // Note that this metric says a sync was enqueued, but once implemented it could be a duplicate so will need to be changed to be more like the log in the if block above this
-      resultBySecondary[secondary] =
-        'new_sync_request_enqueued_secondary_to_primary'
-    } else if (syncMode === SYNC_MODES.None) {
-      resultBySecondary[secondary] = 'no_sync_secondary_data_matches_primary'
+    outcomesBySecondary[secondary] = {
+      syncMode,
+      result
     }
+    continue
```

Comment:
nit: what's the purpose of this continue if there's nothing after it?

Reply:
none - i wasn't sure best practice here, i can remove it tho